src/util/http_utils.js

```diff
@@ -287,19 +295,66 @@ function send_reply(req, res, reply, options) {
     res.end();
 }
 
+/**
+ * Check if a hostname should be proxied or not
+ */
+function should_proxy(hostname) {
+    for (const addr of no_proxy_list) {
+        // Cover the case where the address is an IP
+        if ((ip.isV4Format(addr) || ip.isV6Format(addr)) && ip.isEqual(hostname, addr)) {
+            return false;
+        }
+        // Cover the case where the address is a CIDR subnet
+        try {
+            if (ip.cidrSubnet(addr).contains(hostname)) {
+                return false;
+            }
```

Comment:
First, The try catch block is here to protect against the case where  `addr` is not in the cidr subnet format `#.#.#.#/#` not because of the format of `hostname`

Second, if the hostname does not match the cidr then we still must no return (even if the `addr` is a cidr because we still need to check against the other addresses in the list